wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsCloudClient.java

```diff
@@ -0,0 +1,284 @@
+    {
+        return new HPCCWsCloudClient(connection);
+    }
+
+    /**
+     * Gets a HPCCWsCloudClient based on user provided connection attributes.
+     *
+     * @param protocol
+     *            the protocol
+     * @param targetHost
+     *            the target host
+     * @param targetPort
+     *            the target port
+     * @param user
+     *            the user
+     * @param pass
+     *            the pass
+     * @return the HPCC HPCCWsCloudClient
+     */
+    public static HPCCWsCloudClient get(String protocol, String targetHost, String targetPort, String user, String pass)
+    {
+        Connection conn = new Connection(protocol, targetHost, targetPort);
+        conn.setCredentials(user, pass);
+        return new HPCCWsCloudClient(conn);
+    }
+
+    /**
+     * Gets a HPCCWsCloudClient based on user provided connection attributes.
+     *
+     * @param protocol
+     *            the protocol
+     * @param targetHost
+     *            the target host
+     * @param targetPort
+     *            the target port
+     * @param user
+     *            the user
+     * @param pass
+     *            the pass
+     * @param timeout
+     *            the timeout
+     * @return the HPCC HPCCWsCloudClient
+     */
+    public static HPCCWsCloudClient get(String protocol, String targetHost, String targetPort, String user, String pass, int timeout)
+    {
+        Connection conn = new Connection(protocol, targetHost, targetPort);
+        conn.setCredentials(user, pass);
+        conn.setConnectTimeoutMilli(timeout);
+        conn.setSocketTimeoutMilli(timeout);
+
+        return new HPCCWsCloudClient(conn);
+    }
+
+    /**
+     * Instantiates a new HPCC WSCloud client.
+     *
+     * @param baseConnection
+     *            the base connection
+     */
+    protected HPCCWsCloudClient(Connection baseConnection)
+    {
+        initWsCloudClientStub(baseConnection);
+    }
+
+    /**
+     * Initializes the service's underlying stub Should only be used by constructors.
+     *
+     * @param conn
+     *            -- All connection settings included
+     */
+    protected void initWsCloudClientStub(Connection conn)
+    {
+        setActiveConnectionInfo(conn);
+        try
+        {
+            stub = setStubOptions(new WsCloudStub(conn.getBaseUrl() + WSCLOUDURI), conn);
+        }
+        catch (Exception e)
+        {
+            stub = null;
+
+            initErrMessage = "Could not initialize WsCloudClient - Review all HPCC connection values";
+            if (!e.getLocalizedMessage().isEmpty())
+            {
+                initErrMessage += "\n" + e.getLocalizedMessage();
+            }
+            log.error(initErrMessage);
+        }
+    }
+
+    /**
+     * <p>getPods.</p>
+     *
+     * @return a JSON formatted list of available pods - in raw k8s format.
+     * @throws java.lang.Exception if any.
+     */
+    public String getPods() throws Exception
```

Comment:
Returning raw k8s format here is interesting; I don't think we should take on providing consistency here, but it does lead me to wonder what the use cases for this are?

Reply:
Agreed, I'm a bit surprised the ESP service was designed this way as well, typically they provide well defined response structures.
The use case is likely related to the new ECLWatch interface which lists hpcc cluster pods, and also uses those pod names as potential filters in the new log viewer.
As far as wsclient, this is added for completeness.